src/gluonnlp/embedding/token_embedding.py

```diff
@@ -1019,10 +1019,103 @@ class Word2Vec(TokenEmbedding):
+            vocab_size, vec_len = (int(x) for x in header.split())
+            if self.unknown_token:
+                # Reserve a vector slot for the unknown token at the very beggining
+                # because the unknown token index is 0.
+                all_elems.extend([0] * vec_len)
+            binary_len = np.dtype(np.float32).itemsize * vec_len
+            for line_num in range(vocab_size):
+                token = []
+                while True:
+                    ch = f.read(1)
+                    if ch == b' ':
+                        break
+                    if ch == b'':
+                        raise EOFError('unexpected end of input; is count incorrect or file '
+                                       'otherwise damaged?')
+                    if ch != b'\n':  # ignore newlines in front of words (some binary files have)
+                        token.append(ch)
+                try:
+                    token = b''.join(token).decode(encoding=encoding)
+                except ValueError:
+                    warnings.warn('line {} in {}: failed to decode. Skipping.'
+                                  .format(line_num, pretrained_file_path))
+                    continue
+                elems = np.fromstring(f.read(binary_len), dtype=np.float32)
+
+                assert len(elems) > 1, 'line {} in {}: unexpected data format.'.format(
+                    line_num, pretrained_file_path)
+
+                if token == self.unknown_token and loaded_unknown_vec is None:
+                    loaded_unknown_vec = elems
+                    tokens.add(self.unknown_token)
+                elif token in tokens:
+                    warnings.warn('line {} in {}: duplicate embedding found for '
+                                  'token "{}". Skipped.'.format(line_num, pretrained_file_path,
+                                                                token))
+                else:
+                    assert len(elems) == vec_len, \
+                        'line {} in {}: found vector of inconsistent dimension for token ' \
+                        '"{}". expected dim: {}, found: {}'.format(line_num,
+                                                                   pretrained_file_path,
+                                                                   token, vec_len, len(elems))
+                    all_elems.extend(elems)
+                    self._idx_to_token.append(token)
+                    self._token_to_idx[token] = len(self._idx_to_token) - 1
+                    tokens.add(token)
+        self._idx_to_vec = nd.array(all_elems).reshape((-1, vec_len))
+
+        if self.unknown_token:
+            if loaded_unknown_vec is None:
+                self._idx_to_vec[C.UNK_IDX] = self._init_unknown_vec(shape=vec_len)
+            else:
+                self._idx_to_vec[C.UNK_IDX] = nd.array(loaded_unknown_vec)
+
+    @classmethod
+    def from_w2v_binary(cls, pretrained_file_path, encoding='utf8'):
```

Comment:
I think this can be deleted, given that `__init__` now handles loading binary files? If so, the doc of `__init__` still needs to be updated

Reply:
Keeping this might not be a bad idea. `from_w2v_binary` is align with `from_file`. Both load pre-trained model outside from the source list. Despite that the `__init__` constructor handle binary files, having this function has a similar user interface for user to load pretrained models outside from our source list. 
How do you think? 

Thanks for your review. I updated the doc as well.

Reply:
Thanks @imgarylai! I'm fine with keeping it. While it duplicates functionality indeed it may be easier to discover the functionality for some users.

In that case, can you update the doc to clarify that this is equivalent to `__init__`? Otherwise users may wonder about the difference. Regarding the doc of `__init__`, 

> The name of the pre-trained token embedding file.

is not very clear. Can you clarify that either the name of one of the included embeddings, or a path to a binary file can be provided? Thank you!

Reply:
@leezu Yes, sure!
The doc has been updated.